receiver/dotnetdiagnosticsreceiver/binary/reader_wrapper.go

```diff
@@ -0,0 +1,147 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package binary
+
+import (
+	enc "encoding/binary"
+	"errors"
+)
+
+type ReaderWrapper struct {
+	br    *LinkedBlockingReader
+	order enc.ByteOrder
+	errs  []error
+}
+
+func NewReaderWrapper(br *LinkedBlockingReader) *ReaderWrapper {
+	return &ReaderWrapper{
+		br:    br,
+		order: enc.LittleEndian,
+	}
+}
+
+// TODO handle proper UTF16
+func (r *ReaderWrapper) ReadDoubleByteASCII() (s string, err error) {
+	var b1, b2 byte
+	for {
+		b1 = r.ReadByte()
+		b2 = r.ReadByte()
+		if b1 == 0 && b2 == 0 {
+			return
+		}
+		s += string(b1)
+	}
+}
+
+var singleByte = make([]byte, 1)
+
+//goland:noinspection GoStandardMethods
+func (r *ReaderWrapper) ReadByte() byte {
+	r.Read(singleByte)
+	return singleByte[0]
+}
+
+func (r *ReaderWrapper) AssertNextByteEquals(b byte) error {
+	// TODO append error?
+	if r.ReadByte() != b {
+		return errors.New("next byte assertion failed")
+	}
+	return nil
+}
+
+func (r *ReaderWrapper) ReadASCII(strlen int) string {
+	b := make([]byte, strlen)
+	_, err := r.br.Read(b)
```

Comment:
In general I gather your take on nil checks is that they're unnecessary as long as the constructor/creator helper is always being used?  Checking before each `r.br` proxied call could get tedious.